apps/vc-api/src/key/key.service.ts

```diff
@@ -85,32 +87,77 @@ export class KeyService implements IGenerateKey {
     if (key.privateKey.crv !== 'Ed25519' || key.publicKey.crv !== 'Ed25519') {
       throw new BadRequestException('Only Ed25519 keys are supported');
     }
+
     const privateKey = await importJWK(key.privateKey, ED25519_ALG);
     const publicKey = await importJWK(key.publicKey, ED25519_ALG);
     if ('type' in privateKey && 'type' in publicKey) {
+      const publicKeyBase58 = Base64ToBase58(key.publicKey.x);
+      const keyExists = await this.fetchKey(publicKeyBase58);
+
+      // if key already exists in the wallet
+      // returns keyId i.e. publicKeyBase58
+      if(keyExists != null) {
+        return {
+          keyId: publicKeyBase58,
+        };
+      }
+
       return await this.saveNewKey({ privateKey, publicKey });
     }
     throw new Error(`importJWK produced incorrect type. public key: ${publicKey}`);
   }
 
   public async exportKey(keyDescription: KeyDescriptionDto): Promise<KeyPairDto> {
-    const keyPair = await this.keyRepository.findOneBy({ publicKeyThumbprint: keyDescription.keyId });
-    return keyPair;
+    const key = await this.fetchKey(keyDescription.keyId);
+    if (key) {
+      return {
+        publicKey: {
+          ...key.key.jwkPublic,
+          kid: keyDescription.keyId,
+        },
+        privateKey:
+        {
+          ...key.key.jwkSecret
+        }
+      }
+    }
+    return null;
   }
 
   private async saveNewKey(keyGenResult: GenerateKeyPairResult): Promise<IKeyDescription> {
+
     const publicKeyJWK = await exportJWK(keyGenResult.publicKey);
-    publicKeyJWK.kid = await calculateJwkThumbprint(publicKeyJWK, 'sha256');
     const privateKeyJWK = await exportJWK(keyGenResult.privateKey);
-    const keyEntity = this.keyRepository.create({
-      publicKey: publicKeyJWK,
-      privateKey: privateKeyJWK,
-      publicKeyThumbprint: publicKeyJWK.kid
-    });
-    await this.keyRepository.save(keyEntity);
-    return {
-      keyId: publicKeyJWK.kid
+
+    const privateAskarJwk = Jwk.fromJson(JSON.parse(JSON.stringify(privateKeyJWK)));
```

Comment:
Is it fair to say that the `JSON.parse` and `JSON.stringify` are only used here to remove the type information and allow for TS compilation? If so, I think it is clear to simply cast to `any`.
```suggestion
    const privateAskarJwk = Jwk.fromJson(privateKeyJWK as any);
```

Of course, it would be better if we could avoid the use of `any`. When I tried the above, I get the following type error 
```
Argument of type 'JWK' is not assignable to parameter of type 'JwkProps'.
  Property 'kty' is optional in type 'JWK' but required in type 'JwkProps'.ts(2345)
```
I think we should set the `kty` property in the same way that it is done in the `DIDKeyFactory`: https://github.com/openwallet-foundation-labs/vc-api/blob/develop/apps/vc-api/src/did/did.service.ts#L55-L56
It might not perfect as I'm not sure the type-checking is really working   (as it seems to compile even when I comment out the guard clause 🤔 ), but I think it is safer and clear.
```suggestion
// Need to check that the properties required by Askar JWK are there
if (!privateKeyJWK.crv || !privateKeyJWK.x) {
      throw new Error("Missing required properties from private JWK");
    }
    const jwkProps = { ...privateKeyJWK, kty: 'OKP', crv: privateKeyJWK.crv, x: privateKeyJWK.x };
    const privateAskarJwk = Jwk.fromJson(jwkProps);
```